DQM/CTPPS/plugins/CTPPSRandomDQMSource.cc

```diff
@@ -0,0 +1,201 @@
+/******************************************
+ *
+ * This is a part of CTPPSDQM software.
+ * Authors:
+ *   A. Bellora (Universita' e INFN Torino)
+ *
+ *******************************************/
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+#include "DQMServices/Core/interface/DQMStore.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include "CondFormats/PPSObjects/interface/CTPPSPixelIndices.h"
+#include "DataFormats/CTPPSDetId/interface/CTPPSDetId.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDataError.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelCluster.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSPixelLocalTrack.h"
+#include "DataFormats/Common/interface/TriggerResults.h"
+
+#include <string>
+
+//-----------------------------------------------------------------------------
+
+class CTPPSRandomDQMSource : public DQMEDAnalyzer {
+public:
+  CTPPSRandomDQMSource(const edm::ParameterSet &ps);
+  ~CTPPSRandomDQMSource() override;
+
+protected:
+  void dqmBeginRun(edm::Run const &, edm::EventSetup const &) override;
+  void bookHistograms(DQMStore::IBooker &, edm::Run const &, edm::EventSetup const &) override;
+  void analyze(edm::Event const &e, edm::EventSetup const &eSetup) override;
+
+private:
+  edm::EDGetTokenT<edm::DetSetVector<CTPPSPixelDigi>> tokenDigi;
+
+  static constexpr int NArms = 2;
+  static constexpr int NStationMAX = 3;  // in an arm
+  static constexpr int NRPotsMAX = 6;    // per station
+  static constexpr int NplaneMAX = 6;    // per RPot
+  static constexpr int RPn_first = 3, RPn_last = 4;
+  static constexpr int StationIDMAX = 4;  // possible range of ID
+  static constexpr int RPotsIDMAX = 8;    // possible range of ID
+
+  unsigned int rpStatusWord = 0x8008;      // 220_fr_hr(stn2rp3)+ 210_fr_hr
+  int RPstatus[StationIDMAX][RPotsIDMAX];  // symmetric in both arms
+  int StationStatus[StationIDMAX];         // symmetric in both arms
+  const int IndexNotValid = 0;
+
+  MonitorElement *hBX;
+
+  static constexpr int RPotsTotalNumber = NArms * NStationMAX * NRPotsMAX;
+
+  int RPindexValid[RPotsTotalNumber];
+  MonitorElement *h2HitsVsBXRandoms[RPotsTotalNumber];
+
+  int getRPindex(int arm, int station, int rp) {
```

Comment:
```suggestion
  int getRPindex(int arm, int station, int rp) const {
```
It does not modify any class member.

Reply:
Added.